core/extensions/impl/io_extension.cpp

```diff
@@ -18,13 +19,54 @@ namespace kagome::extensions {
     logger_->info("hex value: {}", buf.toHex());
   }
 
+  void IOExtension::ext_logging_log_version_1(
+      runtime::WasmEnum level,
+      runtime::WasmSpan target,
+      runtime::WasmSpan message) {
+    using runtime::WasmResult;
+    using runtime::WasmLogLevel;
+
+    auto read_str_from_position = [&] (WasmResult location) {
+      return memory_->loadStr(location.address, location.length);
+    };
+
+    const auto target_str = read_str_from_position(WasmResult(target));
+    const auto message_str = read_str_from_position(WasmResult(message));
+
+    switch (level) {
+      case WasmLogLevel::WasmLL_Error:
+        logger_->error(
+            "target: {}\n\tmessage: {}", level, target_str, message_str);
+        break;
+      case WasmLogLevel::WasmLL_Warn:
+        logger_->warn(
+            "target: {}\n\tmessage: {}", level, target_str, message_str);
+        break;
+      case WasmLogLevel::WasmLL_Info:
+        logger_->info(
+            "target: {}\n\tmessage: {}", level, target_str, message_str);
+        break;
+      case WasmLogLevel::WasmLL_Debug:
+        logger_->debug(
+            "target: {}\n\tmessage: {}", level, target_str, message_str);
+        break;
+      case WasmLogLevel::WasmLL_Trace:
+        logger_->trace(
+            "target: {}\n\tmessage: {}", level, target_str, message_str);
+        break;
+      default: {
+        assert(false);
```

Comment:
I guess we should not crash here (at wasm problem). Suggestion: logging represented message any way, and additional record about invalid log-level.

Reply:
There will be no crash here. You've got ```int 3``` only in debug build. In release nothing will happen. Message will be skipped. By maybe you are wright, it will be better to write it anyway.

Reply:
Agree about debug build (but we got SIGABRT). But logging would be usefull anyway. =)